backend/app/adapter/db/change_log.go

```diff
@@ -0,0 +1,101 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/short-d/short/app/adapter/db/table"
+
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+var _ repository.ChangeLog = (*ChangeLogSQL)(nil)
+
+// ChangeLogSQL accesses ChangeLog information in change_log table through SQL.
+type ChangeLogSQL struct {
+	db *sql.DB
+}
+
+// GetChangeByID finds Change from change_log table with given ID.
+func (c ChangeLogSQL) GetChangeByID(id string) (entity.Change, error) {
+	statement := fmt.Sprintf(`SELECT "%s","%s","%s","%s" FROM "%s" WHERE "%s"=$1;`,
+		table.ChangeLog.ColumnID,
+		table.ChangeLog.ColumnTitle,
+		table.ChangeLog.ColumnSummaryMarkdown,
+		table.ChangeLog.ColumnReleasedAt,
+		table.ChangeLog.TableName,
+		table.ChangeLog.ColumnID,
+	)
+
+	row := c.db.QueryRow(statement, id)
+	var change = entity.Change{}
+
+	err := row.Scan(&change.ID, &change.Title, &change.SummaryMarkdown, &change.ReleasedAt)
+	return change, err
+}
+
+// GetChangeLog retrieves full changelog from change_log table.
+func (c ChangeLogSQL) GetChangeLog() ([]entity.Change, error) {
+	statement := fmt.Sprintf(`SELECT "%s","%s","%s","%s" FROM "%s";`,
+		table.ChangeLog.ColumnID,
+		table.ChangeLog.ColumnTitle,
+		table.ChangeLog.ColumnSummaryMarkdown,
+		table.ChangeLog.ColumnReleasedAt,
+		table.ChangeLog.TableName,
+	)
+
+	var changeLog []entity.Change
+	rows, err := c.db.Query(statement)
+
+	if err != nil {
+		return changeLog, err
+	}
+
+	for rows.Next() {
+		change := entity.Change{}
+		err = rows.Scan(&change.ID, &change.Title, &change.SummaryMarkdown, &change.ReleasedAt)
+		if err != nil {
+			return changeLog, err
+		}
+		change.ReleasedAt = utc(change.ReleasedAt)
+		changeLog = append(changeLog, change)
+	}
+
+	return changeLog, nil
+}
+
+// CreateChange adds a new Change into change_log table.
+func (c ChangeLogSQL) CreateChange(newChange entity.Change) (entity.Change, error) {
+	statement := fmt.Sprintf(`
+INSERT INTO "%s" ("%s", "%s","%s","%s")
+VALUES ($1, $2, $3, $4);
+`,
+		table.ChangeLog.TableName,
+		table.ChangeLog.ColumnID,
+		table.ChangeLog.ColumnTitle,
+		table.ChangeLog.ColumnSummaryMarkdown,
+		table.ChangeLog.ColumnReleasedAt,
+	)
+
+	_, err := c.db.Exec(
+		statement,
+		newChange.ID,
+		newChange.Title,
+		newChange.SummaryMarkdown,
+		newChange.ReleasedAt,
+	)
+
+	if err != nil {
+		return entity.Change{}, err
+	}
+
+	return c.GetChangeByID(newChange.ID)
```

Comment:
What is the purpose to make this query if we already saved the entity and know exactly how it should look?

Reply:
I just wanted to be sure that we inserted the change in change_log table in case db.Exec fails without any error... i don't think it will happen though

Reply:
I think it is a redundant check. It is an issue of DB driver, not ours.

Reply:
@byliuyang what's your opinion, shall we remove GetChangeByID method?

Reply:
Please replace this one with the struct we passed in as @alldroll suggested for now.

Reply:
Will do that now